### PR TITLE
Add zamp

### PR DESCRIPTION
**Pull request by vicasze**
Hello, 
this PR is to add zAMP to bioconda. 

[zAMP](https://github.com/metagenlab/zAMP) is a snakemake CLI pipeline designed for convenient, reproducible and scalable amplicon-based metagenomics. 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by vicasze** [comment]
I'd like to reopen the pull request after fixing the version needed for a required package

File: recipes/zamp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 366d1a1bdff60e1a8100298608f20ab2294ef476a1da1bee9c1975641c495f6e
+
+build:
+  number: 0
+  noarch: python
+  entry_points:
+    - zamp=zamp.__main__:main
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --no-cache-dir -vvv
+  run_exports:
+    - {{ pin_subpackage('zamp', max_pin="x.x") }}
```
Comment (martin-g) [inline recipes/zamp/meta.yaml:19]: Usually versions `1.+` should use `max_pin="x"`. Are you sure about `"x.x"` ?
See https://github.com/bioconda/bioconda-recipes/blob/master/.github/PULL_REQUEST_TEMPLATE.md#instructions-for-avoiding-api-abi-and-cli-breakage-issues

**Review by martin-g — APPROVED** [review]

File: recipes/zamp/meta.yaml (hunk 1 above)
Comment (vicasze) [reply recipes/zamp/meta.yaml]: Thank you. I modified meta.yaml to max_pin="x" as suggested.

**Comment by vicasze** [comment]
@BiocondaBot please add label

**Review by martin-g — APPROVED** [review]

**Pull request merged**